pkg/asset/manifests/powervs/cloudproviderconfig.go

```diff
@@ -0,0 +1,79 @@
+package powervs
+
+import (
+	"bytes"
+	"strings"
+	"text/template"
+)
+
+// https://github.com/kubernetes/kubernetes/blob/368ee4bb8ee7a0c18431cd87ee49f0c890aa53e5/staging/src/k8s.io/legacy-cloud-providers/gce/gce.go#L188
+type config struct {
+	Global     global     `gcfg:"global"`
+	Kubernetes kubernetes `gcfg:"kubernetes"`
+	Provider   provider   `gcfg:"provider"`
+}
+
+type global struct {
+	Version string `gcfg:"version"`
+}
+
+type kubernetes struct {
+	ConfigFile string `gcfg:"config-file"`
+}
+
+type provider struct {
+	AccountID                string `gcfg:"accountID"`
+	ClusterID                string `gcfg:"clusterID"`
+	ClusterDefaultProvider   string `gcfg:"cluster-default-provider"`
+	Region                   string `gcfg:"region"`
+	G2CredentialsFilePath    string `gcfg:"g2Credentials"`
+	G2ResourceGroupName      string `gcfg:"g2ResourceGroupName"`
+	G2VPCName                string `gcfg:"g2VpcName"`
+	G2WorkerServiceAccountID string `gcfg:"g2workerServiceAccountID"`
+	G2VPCSubnetNames         string `gcfg:"g2VpcSubnetNames"`
+}
+
+// CloudProviderConfig generates the cloud provider config for the IBMCloud platform.
+func CloudProviderConfig(infraID string, accountID string, vpcName string, region string, resourceGroupName string, subnets []string) (string, error) {
```

Comment:
can this not be common with ibmcloud? i don't see any difference other than the subnets?

Reply:
i believe @Karthik-K-N said they'll be adding some additional items when they get the ccm fork available, so i'd defer to him on that. but if we do agree to that let's make a tracker so we don't misplace that thought. in the meantime, ok to merge this as-is?

Reply:
if there's going to be some deviation then yeah, it's ok

Reply:
also a nit to change the above comment from IBMCloud to PowerVS